### PR TITLE
feat: add `array/base/broadcasted-ternary4d`

### LINKED ISSUE #3165: [RFC]: add `array/base/broadcasted-ternary4d`
### Description

This RFC proposes adding a new package to the `array/base/*` namespace: `@stdlib/array/base/broadcasted-ternary4d`.

This package is the ternary equivalent of [`@stdlib/array/base/broadcasted-binary4d`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/broadcasted-binary4d).

To provide a concrete example of what a PR adding the desired package should contain, see [`@stdlib/array/base/broadcasted-ternary2d`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/broadcasted-ternary2d), which is the 2D package equivalent.

While the package proposed in this RFC will **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages.

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by Vinit-Pandit**
Resolves #3165 .

## Description

> What is the purpose of this pull request?

This pull request:

-   adding array/base/broadcasted-ternary4d

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3165 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

-   [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)
* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+var shapes = [
+    [ 1, 1, 1 ],
+    [ 1, 1, 2 ],
+    [ 1, 1, 1, 2 ],
+    [ 1, 1, 2, 2 ]
+];
+
+bternary4d( [ x, y, z, out ], shapes, add );
+// out => [ [ [ [ 7.0 ,9.0 ], [ 7.0 ,9.0 ] ] ] ]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/README.md]:
```suggestion
// out => [ [ [ [ 7.0, 9.0 ], [ 7.0, 9.0 ] ] ] ]
```

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+Applies a ternary callback to elements in three [broadcasted][@stdlib/array/base/broadcast-array] nested input arrays and assigns results to elements in a four-dimensional nested output array.
+
+```javascript
+var add = require( '@stdlib/math/base/ops/add3' );
+var zeros4d = require( '@stdlib/array/base/zeros4d' );
+
+var x = [ [ [ 1.0 ] ] ];
+var y = [ [ [ 2.0, 3.0 ] ] ];
+var z = [ [ [ [4.0, 5.0] ] ] ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/README.md]:
```suggestion
var z = [ [ [ [ 4.0, 5.0 ] ] ] ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    fcn: Function
+        Ternary callback.
+
+    Examples
+    --------
+    > function fcn( x, y, z ) { return x + y + z };
+    > var x = [ [ [ 1.0 ] ] ];
+    > var y = [ [ [ 2.0 ,  3.0 ] ] ];
+    > var z = [ [ [ [4.0 ,5.0] ] ] ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt]:
```suggestion
    > var z = [ [ [ [ 4.0, 5.0 ] ] ] ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        Array shapes.
+
+    fcn: Function
+        Ternary callback.
+
+    Examples
+    --------
+    > function fcn( x, y, z ) { return x + y + z };
+    > var x = [ [ [ 1.0 ] ] ];
+    > var y = [ [ [ 2.0 ,  3.0 ] ] ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt]:
```suggestion
    > var y = [ [ [ 2.0, 3.0 ] ] ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    > var z = [ [ [ [4.0 ,5.0] ] ] ];
+    > var out = [ [ [ [ 0.0 ,0.0 ], [ 0.0 ,0.0 ] ] ] ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt]:
```suggestion
    > var out = [ [ [ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ] ] ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    > var shapes =[ [ 1, 1, 1 ], [ 1, 1, 2 ], [ 1, 1, 1, 2 ], [ 1, 1, 2, 2 ] ];
+    > {{alias}}( [ x, y, z, out ], shapes, fcn );
+    > out
+    [ [ [ [ 7.0 ,9.0 ], [ 7.0 ,9.0 ] ] ] ]
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt]: @MeastroZI I recommend doing a very close (i.e., line-by-line) review of your PR to catch all the little mistakes regarding formatting.

Comment (Vinit-Pandit) [reply lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/repl.txt]: Noted !

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/types/test.ts [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+	const z = [
+		[ [1.0, 2.0], [ 3.0, 4.0 ] ], [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ],
+		[ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ], [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ]
+	];
+	const out = [
+		[ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ], [ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ],
+		[ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ], [ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ]
+	];
+
+	const shapes: [ Array<number>, Array<number>, Array<number>, Array<number> ] = [ [ 2, 2, 2, 2 ], [ 2, 2, 2, 2 ], [ 2, 2, 2, 2 ], [ 2, 2, 2, 2 ]];
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/types/test.ts]:
```suggestion
	const shapes: [ Array<number>, Array<number>, Array<number>, Array<number> ] = [ [ 2, 2, 2, 2 ], [ 2, 2, 2, 2 ], [ 2, 2, 2, 2 ], [ 2, 2, 2, 2  ] ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/types/test.ts [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+// The compiler throws an error if the function is provided a second argument which is not an array of arrays...
+{
+	const x = [
+		[ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ], [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ],
+		[ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ], [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ]
+	];
+	const y = [
+		[ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ], [ [ 1.0, 2.0 ], [3.0, 4.0]],
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/docs/types/test.ts]:
```suggestion
		[ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ], [ [ 1.0, 2.0 ], [ 3.0, 4.0 ] ],
```

File: lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/lib/index.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+*
+* var x = ones4d( shapes[ 0 ] );
+* var y = ones4d( shapes[ 1 ] );
+* var z = ones4d( shapes[ 2 ] );
+* var out = zeros4d( shapes[ 3 ] );
+*
+* bternary3d( [ x, y, z, out ], shapes, add );
+*
+* console.log( out );
+* // => [ [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ] , [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ], [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ] , [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ] ]
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/array/base/broadcasted-ternary4d/lib/index.js]:
```suggestion
* // => [ [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ], [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ], [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ], [ [ 3.0, 3.0 ], [ 3.0, 3.0 ] ] ]
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks for working on this, @MeastroZI.

Quick note: please make sure you have properly configured your local development environment according to the project dev guide. I would have expected our Git hooks to catch various lint errors observed in the TypeScript test files. Before every push, linting should be automatically run. If that isn't happening, then you need to run `make init`.

**Pull request merged**